### PR TITLE
CONTRIBUTION.md and README.md update

### PR DESCRIPTION
**Pull request by yhontyk**
### What does this PR do?
Creates a new and rehashed contribution guide. 

### Specify the version of the product this PR applies to.

### CODE REVIEW

File: README.md
Comment (kowen-rh) [inline README.md]: "The" is not needed before "Eclipse Che".

"See the Contribution guide". Personally, though, I'd rewrite this sentence, maybe something like "to contribute to the documentation, see the Contribution guide."

File: CONTRIBUTION.md
Comment (kowen-rh) [inline CONTRIBUTION.md]: Is there a reason this sentence is continued on another line?

File: CONTRIBUTION.md
Comment (kowen-rh) [inline CONTRIBUTION.md]: "...we as contributors and maintainers pledge to make participation..."

"Make" instead of "making", basically. :)

File: CONTRIBUTION.md
Comment (kowen-rh) [inline CONTRIBUTION.md]: I'd use "or" instead of the slash here.

File: CONTRIBUTION.md
Comment (kowen-rh) [inline CONTRIBUTION.md]: Extraneous new line?

File: CONTRIBUTION.md
Comment (kowen-rh) [inline CONTRIBUTION.md]: Extraneous new line?

File: CONTRIBUTION.md
Comment (kowen-rh) [inline CONTRIBUTION.md:31]: Extraneous new line?

File: CONTRIBUTION.md
Comment (kowen-rh) [inline CONTRIBUTION.md]: These new lines are very inconsistent. Are you trying to adhere to a specific column width? If so, new lines need to be applied more consistently.

Personally, I'd keep each sentence or paragraph contiguous.

File: CONTRIBUTION.md
Comment (kowen-rh) [inline CONTRIBUTION.md]: "submit the PR against the `master` branch"

File: CONTRIBUTION.md
Comment (kowen-rh) [inline CONTRIBUTION.md]: "The Eclipse Che documentation projects follows a (or "the"? I'm not sure, honestly) forking workflow."

"Fork the repository and ensure you create all branches in your own fork before you start working." -- This just seems more active to me. :)

File: CONTRIBUTION.md
Comment (kowen-rh) [inline CONTRIBUTION.md]: Might want to comment this section out if it's a WIP. :)

File: CONTRIBUTION.md
Comment (kowen-rh) [inline CONTRIBUTION.md]: Technically, I'd say AsciiDoc is used for markup, not formatting. (Also, "uses" instead of "use" here.)

File: CONTRIBUTION.md
Comment (kowen-rh) [inline CONTRIBUTION.md]: "use the `test-adoc` script"

File: CONTRIBUTION.md
Comment (kowen-rh) [inline CONTRIBUTION.md]: Is it `test-adoc.sh` or just `test-adoc`?

File: CONTRIBUTION.md
Comment (kowen-rh) [inline CONTRIBUTION.md:63]: Same, which is it? :)

File: CONTRIBUTION.md
Comment (kowen-rh) [inline CONTRIBUTION.md]: "The Eclipse Che documentation project follows guidelines from the Modular Documentation initiative." (Also, is modular documentation actually capitalized? I'm not sure.)

File: CONTRIBUTION.md
Comment (kowen-rh) [inline CONTRIBUTION.md]: I'm honestly not sure if we should use "a" or "an" here, because when I read this, I hear "create an dot-adoc file" in my head. ;)

File: CONTRIBUTION.md
Comment (kowen-rh) [inline CONTRIBUTION.md]: Is this "a `newdoc`" script or "the `newdoc`" script? I don't know how this is implemented, so I thought it was worth asking. :)

File: CONTRIBUTION.md
Comment (kowen-rh) [inline CONTRIBUTION.md]: Double "the".

File: CONTRIBUTION.md
Comment (kowen-rh) [inline CONTRIBUTION.md]: Another awkward line break.

File: CONTRIBUTION.md
Comment (kowen-rh) [inline CONTRIBUTION.md]: "defined in the Eclipse Che project"

Also, should this be "the Eclipse Che documentation project"?

File: CONTRIBUTION.md
Comment (pmkovar) [inline CONTRIBUTION.md:1]: Just out of curiosity, is this generated by a tool?

File: CONTRIBUTION.md
Comment (pmkovar) [inline CONTRIBUTION.md]: I would suggest that the ToC follows a typical workflow, so talk about Creating documentation first, then building, then submitting a PR, then publishing.

File: CONTRIBUTION.md
Comment (pmkovar) [inline CONTRIBUTION.md]: Can we add a short section on reviewing? I'd particularly like to call out that we need procedural PRs to get tested by QE before merging.

File: README.md
Comment (pmkovar) [inline README.md:5]: Maybe emphasize that `eclipse/che/issues` is not a place for support questions? Unless it is.

File: CONTRIBUTION.md
Comment (yhontyk) [reply CONTRIBUTION.md:1]: I used this script -  https://github.com/ekalinin/github-markdown-toc

File: README.md
Comment (yhontyk) [reply README.md:5]: I don't understand. Where else are users expected to ask questions?

Comment (rkratky) [reply README.md:5]: I believe it is. There 'question' label for it.

**Review by rkratky — APPROVED** [review]

**Pull request merged**
